File: src/micapolos/tata8/Flip.java
package micapolos.tata8;

public final class Flip {
  public boolean x;
  public boolean y;

  public void set(boolean x, boolean y) {
    this.x = x;
    this.y = y;
  }

  @Override
  public String toString() {
    return String.format("flip(%s,%s)", x, y);
  }
}

File: src/micapolos/zexy2/Flip.kt
package micapolos.zexy2

import micapolos.zexy2.ast.Expression

class Flip<T>(val x: Expression<T>, val y: Expression<T>)

fun flip(x: Double, y: Double) = Flip(constant(x), constant(y))
fun <T> flip(x: Expression<T>, y: Double) = Flip(x, constant(y))
fun <T> flip(x: Double, y: Expression<T>) = Flip(constant(x), y)
fun <T> flip(x: Expression<T>, y: Expression<T>) = Flip(x, y)